# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioStream.cpp
# Commit: 3fbc1456c6d2
# Full Hash: 3fbc1456c6d26d91729498ea6ad19dc24e479dfa
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2019-10-02 03:38:52
# Regressor Bug: 1544997
# File Overlap Count: 1
# Description:
#   Bug 1544997 - Propogate cubeb stream error to AudioSink to handle loss of audio device.  r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D47745
# ==============================================================================

diff -r f53153babb6f -r 3fbc1456c6d2 dom/media/AudioStream.cpp
--- a/dom/media/AudioStream.cpp	Tue Oct 01 20:01:54 2019 +0000
+++ b/dom/media/AudioStream.cpp	Tue Oct 01 12:14:44 2019 +0000
@@ -604,6 +604,7 @@
   } else if (aState == CUBEB_STATE_ERROR) {
     LOGE("StateCallback() state %d cubeb error", mState);
     mState = ERRORED;
+    mDataSource.Errored();
   }
 }
 